src/benchmarks/micro/corefx/System.Globalization/Perf.StringCultureSpecific.cs

```diff
@@ -0,0 +1,94 @@
+            yield return (new CultureInfo("pl-PL"), CompareOptions.None);
+        }
+
+        [ParamsSource(nameof(GetCultureOptions))]
+        public (CultureInfo CultureInfo, CompareOptions CompareOptions) CultureOptions;
+
+        [Params(
+            128, // small input that fits into stack-allocated array
+            1024 * 128)] // medium size input that fits into an array rented from ArrayPool.Shared without allocation
+        public int Count { get; set; }
+
+        [GlobalSetup]
+        public void Setup()
+        {
+            // we are using part of Alice's Adventures in Wonderland text as test data
+            // it contains mostly simply ascii characters, but also some "high" chars that get special treatment and hit the slow path
+            char[] characters = File.ReadAllText(CompressedFile.GetFilePath("alice29.txt")).ToCharArray().Take(Count).ToArray();
+            _value = new string(characters);
+            _same = new string(characters);
+            _firstHalf = new string(characters.Take(Count / 2).ToArray());
+            _secondHalf = new string(characters.Skip(Count / 2).ToArray());
+            char[] copy = characters.ToArray();
+            copy[0] = (char)(copy[0] + 1);
+            _diffAtFirstChar = new string(copy);
+            copy = characters.ToArray();
+            copy[Count - 1] = (char)(copy[Count - 1] + 1);
+            _diffAtLastChar = new string(copy);
+        }
+
+        [Benchmark]
+        public new void GetHashCode() => CultureOptions.CultureInfo.CompareInfo.GetHashCode(_value, CultureOptions.CompareOptions);
+
+        [Benchmark] // the most work to do: the strings have same conent, but don't point to the same memory
+        public int Compare_Same() => CultureOptions.CultureInfo.CompareInfo.Compare(_value, _same, CultureOptions.CompareOptions);
+
+        [Benchmark] // this should return quickly
+        public int Compare_DifferentFirstChar() => CultureOptions.CultureInfo.CompareInfo.Compare(_value, _diffAtFirstChar, CultureOptions.CompareOptions);
+
+        [Benchmark]
+        public bool IsPrefix_FirstHalf() => CultureOptions.CultureInfo.CompareInfo.IsPrefix(_value, _firstHalf, CultureOptions.CompareOptions);
+
+        [Benchmark] // this should return quickly
+        public bool IsPrefix_DifferentFirstChar() => CultureOptions.CultureInfo.CompareInfo.IsPrefix(_value, _diffAtFirstChar, CultureOptions.CompareOptions);
+
+        [Benchmark]
+        public bool IsSuffix_SecondHalf() => CultureOptions.CultureInfo.CompareInfo.IsSuffix(_value, _secondHalf, CultureOptions.CompareOptions);
+
+        [Benchmark] // this should return quickly
+        public bool IsSuffix_DifferentLastChar() => CultureOptions.CultureInfo.CompareInfo.IsSuffix(_value, _diffAtLastChar, CultureOptions.CompareOptions);
+
+        [Benchmark]
+        public int IndexOf_SecondHalf() => CultureOptions.CultureInfo.CompareInfo.IndexOf(_value, _secondHalf, CultureOptions.CompareOptions);
```

Comment:
The typical IndexOf use searches for small string (e.g. single world) in medium amount of text (e.g. single line).  We should have a coverage for a case like this.

Reply:
another good point, I am going to add such case

Reply:
I've changed the benchmark to search for "word" in ".NET Conf provides a wide selection of live sessions streaming here that feature speakers from the community and .NET product teams. It is a chance to learn, ask questions live, and get inspired for your next software project."